src/app/qgsfeatureaction.cpp

```diff
@@ -53,22 +53,18 @@ QgsAttributeDialog *QgsFeatureAction::newDialog( bool cloneFeature )
 {
   QgsFeature *f = cloneFeature ? new QgsFeature( *mFeature ) : mFeature;
 
-  QgsAttributeEditorContext context;
+  QgsAttributeEditorContext context( QgisApp::instance()->createAttributeEditorContext() );
 
   QgsDistanceArea myDa;
 
   myDa.setSourceCrs( mLayer->crs(), QgsProject::instance()->transformContext() );
   myDa.setEllipsoid( QgsProject::instance()->ellipsoid() );
 
   context.setDistanceArea( myDa );
-  context.setVectorLayerTools( QgisApp::instance()->vectorLayerTools() );
-  context.setMapCanvas( QgisApp::instance()->mapCanvas() );
-  context.setCadDockWidget( QgisApp::instance()->cadDockWidget() );
-  context.setMainMessageBar( QgisApp::instance()->messageBar() );
   context.setFormMode( QgsAttributeEditorContext::StandaloneDialog );
 
   QgsAttributeDialog *dialog = new QgsAttributeDialog( mLayer, f, cloneFeature, parentWidget(), true, context );
-  dialog->setWindowFlags( dialog->windowFlags() | Qt::Tool );
```

Comment:
Is this done somewhere else now?

Reply:
No, it was from a previous modification, I forget to restore it. It's done now.